mlir/test/Conversion/GPUToNVVM/gpu-to-nvvm.mlir

```diff
@@ -917,6 +929,20 @@ gpu.module @test_module_48 {
   }
 }
 
+gpu.module @test_module_49 {
+// CHECK-LABEL: func @explicit_id_bounds()
+  func.func @explicit_id_bounds() -> (index, index, index) {
+    // CHECK: = nvvm.read.ptx.sreg.tid.x range <i32, 0, 32> : i32
+    %0 = gpu.thread_id x upper_bound 32
+    // CHECK: = nvvm.read.ptx.sreg.ntid.x range <i32, 1, 33> : i32
+    %1 = gpu.block_dim x upper_bound 32
```

Comment:
With the given upper-bound 32, shouldn't the check also be for 32 ?
or, is it 33 since its lower-bound starts from 1 ?
or,
May be I am missing something here.. Could you please clarify?

Reply:
The semantics on `upper_bound` are a bit funky: for ID-like ops, it's `0 <= ID < upper_bound` but for dimension numbers, it's `1 <= dim <= upper_bound`

Reply:
I'm open to tweaking that

Reply:
No, that makes sense. For dims, 0 probably is not very meaningful.